proposals/left-right-join-in-query-expressions.md

```diff
@@ -0,0 +1,427 @@
+select «v»
+```
+
+is translated into
+
+```csharp
+( «e1» ) . Join( «e2» , «x1» => «k1» , «x2» => «k2» , ( «x1» , «x2» ) => «v» )
+```
+
+> *Example*: The example
+>
+> ```csharp
+> from c in customersh
+> join o in orders on c.CustomerID equals o.CustomerID
+> select new { c.Name, o.OrderDate, o.Total }
+> ```
+>
+> is translated into
+>
+> ```csharp
+> (customers).Join(
+>    orders,
+>    c => c.CustomerID, o => o.CustomerID,
+>    (c, o) => new { c.Name, o.OrderDate, o.Total })
+> ```
+>
+> *end example*
+
+A `join` clause followed by a query body clause:
+
+```csharp
+from «x1» in «e1»  
+join «x2» in «e2» on «k1» equals «k2»  
+...
+```
+
+is translated into
+
+```csharp
+from * in ( «e1» ) . Join(  
+«e2» , «x1» => «k1» , «x2» => «k2» ,
+( «x1» , «x2» ) => new { «x1» , «x2» })  
+...
+```
+
+A `join`-`into` clause immediately followed by a `select` clause
+
+```csharp
+from «x1» in «e1»  
+join «x2» in «e2» on «k1» equals «k2» into «g»  
+select «v»
+```
+
+is translated into
+
+```csharp
+( «e1» ) . GroupJoin( «e2» , «x1» => «k1» , «x2» => «k2» ,
+                     ( «x1» , «g» ) => «v» )
+```
+
+A `join into` clause followed by a query body clause
+
+```csharp
+from «x1» in «e1»  
+join «x2» in «e2» on «k1» equals «k2» into *g»  
+...
+```
+
+is translated into
+
+```csharp
+from * in ( «e1» ) . GroupJoin(  
+   «e2» , «x1» => «k1» , «x2» => «k2» , ( «x1» , «g» ) => new { «x1» , «g» })
+...
+```
+
+> *Example*: The example
+>
+> ```csharp
+> from c in customers
+> join o in orders on c.CustomerID equals o.CustomerID into co
+> let n = co.Count()
+> where n >= 10
+> select new { c.Name, OrderCount = n }
+> ```
+>
+> is translated into
+>
+> ```csharp
+> from * in (customers).GroupJoin(
+>     orders,
+>     c => c.CustomerID,
+>     o => o.CustomerID,
+>     (c, co) => new { c, co })
+> let n = co.Count()
+> where n >= 10
+> select new { c.Name, OrderCount = n }
+> ```
+>
+> the final translation of which is
+>
+> ```csharp
+> customers
+>     .GroupJoin(
+>         orders,
+>         c => c.CustomerID,
+>         o => o.CustomerID,
+>         (c, co) => new { c, co })
+>     .Select(x => new { x, n = x.co.Count() })
+>     .Where(y => y.n >= 10)
+>     .Select(y => new { y.x.c.Name, OrderCount = y.n })
+> ```
+>
+> where `x` and `y` are compiler generated identifiers that are otherwise invisible and inaccessible.
+>
+> *end example*
+
+</details>
+
+Following is the new proposed dedicated section for the `join` clauses. This section would be numbered as 11.17.3.6 (pushing down the existing sections), or appended at the end. Note that although the `join into` clause is moved into the new section, it is otherwise completely unaffected by this proposal; `join into` translates to `GroupJoin()`, which already performs a conceptual left join (where an outer element has no correlated inner element, it is returned with an empty inner group).
+
+#### 11.17.3.6 Join clauses
+
+##### 11.17.3.6.1 Join clause
```

Comment:
You could state that the section removed above is added verbatim into this section (instead of having to repeat it).